File: Easy/3318. Find X-Sum of All K-Long Subarrays I/Solution.cs
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] FindXSum(int[] nums, int k, int x) {
        int answerSize = nums.Length - k + 1;
        int[] answer = new int[answerSize];

        for (int i = 0; i < answerSize; i++) {
            
            int[] subArray = new int[k];
            for (int j = 0; j < k; j++) {
                subArray[j] = nums[i + j];
            }
            
            int xSum = XSumCalc(subArray, x);
            answer[i] = xSum;
        }
        return answer;
    }

    public int XSumCalc(int[] arr, int x) {
        Dictionary<int, int> freq = new Dictionary<int, int>();

        foreach (int num in arr) {
            if (freq.ContainsKey(num)) {
                freq[num]++;
            } else {
                freq[num] = 1;
            }
        }
        
        var sortedEntries = freq.OrderByDescending(pair => pair.Value)
                                .ThenByDescending(pair => pair.Key)
                                .ToList();
        
        int totalSum = 0;
        
        for (int i = 0; i < x && i < sortedEntries.Count; i++) {
            var entry = sortedEntries[i];
            totalSum += entry.Key * entry.Value;
        }
        
        return totalSum;
    }
}

File: Easy/3318. Find X-Sum of All K-Long Subarrays I/Solution.java
import java.util.HashMap;
import java.util.Map;
import java.util.ArrayList;
import java.util.List;
import java.util.Collections;

class Solution {
    public int[] findXSum(int[] nums, int k, int x) {
        int answerSize = nums.length - k + 1;
        int[] answer = new int[answerSize];

        for (int i = 0; i < answerSize; i++) {
            
            int[] subArray = new int[k];
            for (int j = 0; j < k; j++) {
                subArray[j] = nums[i + j];
            }
            
            int xSum = xSumCalc(subArray, x);
            answer[i] = xSum;
        }
        return answer;
    }

    public int xSumCalc(int[] arr, int x) {
        Map<Integer, Integer> freq = new HashMap<>();

        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }
        
        List<Map.Entry<Integer, Integer>> sortedEntries = new ArrayList<>(freq.entrySet());
        
        Collections.sort(sortedEntries, (a, b) -> {
            if (a.getValue() != b.getValue()) {
                return b.getValue().compareTo(a.getValue());
            } else {
                return b.getKey().compareTo(a.getKey());
            }
        });
        
        int totalSum = 0;
        
        for (int i = 0; i < x && i < sortedEntries.size(); i++) {
            Map.Entry<Integer, Integer> entry = sortedEntries.get(i);
            totalSum += entry.getKey() * entry.getValue();
        }
        
        return totalSum;
    }
}